### PR TITLE
Markov Distribution

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
For the record: 
The Analytical solution will be 
```0.40425532, 0.27659574, 0.31914894```
With 10000 runs, the statistical results are:
```0.3987, 0.2823, 0.3190```

File: framework/Distributions.py [hunk 1]
```diff
@@ -1685,6 +1688,164 @@ def rvs(self):
...
+          self.dataFileType = child.parameterValues['fileType']
+        else:
+          self.dataFileType = 'csv'
+    #Check the correctness of user inputs
+    invalid = self.dataFile is None and self.transition is None
+    if invalid:
+      self.raiseAnError(IOError, "Transition matrix is not provided, please use 'transition' or 'dataFile' node to provide the transition matrix!")
+    invalid = self.dataFile is not None and self.transition is not None
+    if invalid:
+      self.raiseAWarning("Both 'transition' and 'dataFile' nodes are used to provide the transition matrix, only information from 'transition' will be used!")
```
Comment (mandd) [inline framework/Distributions.py]: should this be an error?

File: tests/framework/Distributions/test_distributionsMarkov.xml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      <Input  class="DataObjects"  type="PointSet"      >PointSet_Grid</Input>
+      <Output class="OutStreams"   type="Print"         >MC_dump</Output>
+      <Output class="OutStreams"   type="Print"         >Grid_dump</Output>
+      <Output class="OutStreams"   type="Plot"          >plotXMC</Output>
+      <Output class="OutStreams"   type="Plot"          >plotYMC</Output>
+    </IOStep>
+  </Steps>
+
+  <Models>
+    <ExternalModel ModuleToLoad="simple" name="PythonModule" subType="">
```
Comment (mandd) [inline tests/framework/Distributions/test_distributionsMarkov.xml]: did u upload the python module?

File: framework/Distributions.py (hunk 1 above)
Comment (wangcj05) [reply framework/Distributions.py]: I agree with you ```error``` will be more safe.

File: tests/framework/Distributions/test_distributionsMarkov.xml (hunk 2 above)
Comment (wangcj05) [reply tests/framework/Distributions/test_distributionsMarkov.xml]: I forgot. I have uploaded now.

**Pull request merged**